### PR TITLE
ucx: 1.8.1 -> 1.9.0

### PR DESCRIPTION
**Pull request by markuskowa**
###### Motivation for this change
Regular upgrade:
https://github.com/openucx/ucx/blob/master/NEWS

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by markuskowa** [comment]
@GrahamcOfBorg test slurm

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 105377` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>19 packages marked as broken and skipped:</summary>
  <ul>
    <li>python27Packages.fipy</li>
    <li>python37Packages.arviz</li>
    <li>python37Packages.baselines</li>
    <li>python37Packages.cntk</li>
    <li>python37Packages.dm-sonnet</li>
    <li>python37Packages.edward</li>
    <li>python37Packages.graph_nets</li>
    <li>python37Packages.optuna</li>
    <li>python37Packages.pymc3</li>
    <li>python37Packages.rl-coach</li>
    <li>python37Packages.tensorflow</li>
    <li>python37Packages.tensorflow-build</li>
    <li>python37Packages.tensorflow-build_1</li>
    <li>python37Packages.tensorflow-probability</li>
    <li>python37Packages.tensorflowWithCuda</li>
    <li>python37Packages.tensorflowWithoutCuda</li>
    <li>python37Packages.tensorflow_1</li>
    <li>python37Packages.tflearn</li>
    <li>python38Packages.cntk</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>cntk</li>
    <li>openmolcas</li>
    <li>python27Packages.cntk</li>
    <li>python37Packages.datashader</li>
    <li>python38Packages.datashader</li>
  </ul>
</details>
<details>
  <summary>63 packages built:</summary>
  <ul>
    <li>dl-poly-classic-mpi</li>
    <li>elmerfem</li>
    <li>freecad</li>
    <li>getdp</li>
    <li>globalarrays</li>
    <li>gromacsDoubleMpi</li>
    <li>gromacsMpi</li>
    <li>haskellPackages.haskell-mpi</li>
    <li>hdf5-mpi</li>
    <li>highfive-mpi</li>
    <li>hpcg</li>
    <li>hpl</li>
    <li>ior</li>
    <li>lammps-mpi</li>
    <li>migrate</li>
    <li>netcdf-mpi</li>
    <li>neuron-full (python27Packages.neuron-mpi)</li>
    <li>neuron-mpi</li>
    <li>openmpi</li>
    <li>paraview</li>
    <li>parmetis</li>
    <li>precice</li>
    <li>python27Packages.mpi4py</li>
    <li>python37Packages.dask-gateway</li>
    <li>python37Packages.dask-jobqueue</li>
    <li>python37Packages.dask-ml</li>
    <li>python37Packages.dask-mpi</li>
    <li>python37Packages.dask-xgboost</li>
    <li>python37Packages.dftfit</li>
    <li>python37Packages.distributed</li>
    <li>python37Packages.fenics</li>
    <li>python37Packages.fipy</li>
    <li>python37Packages.h5py-mpi</li>
    <li>python37Packages.lammps-cython</li>
    <li>python37Packages.mask-rcnn</li>
    <li>python37Packages.mpi4py</li>
    <li>python37Packages.neuron-mpi</li>
    <li>python37Packages.streamz</li>
    <li>python37Packages.stumpy</li>
    <li>python37Packages.tensorflow_2 (python37Packages.tensorflow-build_2)</li>
    <li>python38Packages.dask-gateway</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.dask-mpi</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.dftfit</li>
    <li>python38Packages.distributed</li>
    <li>python38Packages.fenics</li>
    <li>python38Packages.fipy</li>
    <li>python38Packages.h5py-mpi</li>
    <li>python38Packages.lammps-cython</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.mpi4py</li>
    <li>python38Packages.neuron-mpi</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.stumpy</li>
    <li>python38Packages.tensorflow_2 (python38Packages.tensorflow-build_2)</li>
    <li>quantum-espresso-mpi</li>
    <li>raxml-mpi</li>
    <li>scalapack</li>
    <li>scotch</li>
    <li>siesta-mpi</li>
    <li>ucx</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
- cntk

Already broken on master:

```
Invalid openblas directory /nix/store/fl9d5425p24capww1f0z6h4mkxdxj8p1-openblas-0.3.12
builder for '/nix/store/jwrq3c1vw3cip8ns2qvdsb99lnywd8br-CNTK-2.7.drv' failed with exit code 1
error: build of '/nix/store/jwrq3c1vw3cip8ns2qvdsb99lnywd8br-CNTK-2.7.drv' failed
```

- openmolcas

source hash mismatch on master

- python27Packages.cntk
- python37Packages.datashader
- python38Packages.datashader

Still please do not self merge without nixpkgs-review to make sure you are not introducing regressions.

**Comment by markuskowa** [comment]
Thanks for double checking but I think in this particular case nixpkgs-review is not necessary. It is only a plugin of openmpi that depends on UCX. I think it sufficient here to check that openmpi works properly. Even though an update of UCX generates a lot of rebuilds it is unlikely that any other package is impacted. 
It is good to know though that openmolcas ran into yet another hash mismatch problem. I will take look at that.
